pkg/diagnostics/diagnostics.go

```diff
@@ -48,11 +55,19 @@ func (r *Registry) RegisterConnectedPeersSource(
 ) {
 	r.Registry.RegisterSource("connected_peers", func() string {
 		connectionManager := netProvider.ConnectionManager()
-		connectedPeers := connectionManager.ConnectedPeers()
+		connectedPeersAddrInfo := connectionManager.ConnectedPeersAddrInfo()
+
+		var peersList []map[string]interface{}
+		for peer, multiaddrs := range connectedPeersAddrInfo {
+			var addresses []string
+			var peerAddr string
+			for _, address := range multiaddrs {
+				// Multiaddr is formatted as follows /<protocol_code>/<address>/<connection_protocol>/<port>
+				// Address is the 1st index
+				peerAddr = strings.Split(strings.Trim(address, "/"), "/")[1]
+				addresses = append(addresses, peerAddr)
```

Comment:
Wouldn't it be better to return the entire multiaddr and let the consuming code extract address, if needed?

I think knowing the port address and protocol code could be beneficial for debugging problems.

Reply:
+1 ^^

Reply:
I *guess* the parsing could also be the reason we sometimes see null addresses [here](https://github.com/keep-network/keep-core/pull/3255#issuecomment-1255898552). Not all cases may be addressed.

Reply:
Sure, we can remove all the filtering from here. It is harder to parse strings in the bash scripts, but we won't touch the client if we'll need some other info from the multiaddrs. [414caae](https://github.com/keep-network/keep-core/pull/3255/commits/414caae76270fe8e549a361de8accf4d00fe9ba4)

Not sure what's the reason for `null`, I can't recreate it
```
{
   "client_info":{
      "chain_address":"0x6357132BA57Ef90b71ffEfdaf0ABDdEA91C07008",
      "network_id":"16Uiu2HAmCrd9nvuum995QoPwdfvTn2siCQtyBkLZnrcYHoLuKnuS",
      "revision":"unknown",
      "version":"unknown"
   },
   "connected_peers":[
      {
         "chain_address":"0xCa8754f7060A0648824f274e3a4d897FA497139D",
         "multiaddrs":[
            "/dns4/keep-validator-0.eks-ap-northeast-2-secure.staging.staked.cloud/tcp/3919"
         ],
         "network_id":"16Uiu2HAm77eSvRq5ioD4J8VFPkq3bJHBEHkssCuiFkgAoABwjo2S"
      },
      {
         "chain_address":"0x1F91c3e91745aB09cc8e619ceD8Fa79B602e3649",
         "multiaddrs":[
            "/ip4/34.141.9.57/tcp/3919"
         ],
         "network_id":"16Uiu2HAky2Y4Tyq5vTA1CxikcDes6o5EH11i2qcg5dBV9W3Lks5c"
      },
      {
         "chain_address":"0x5232E393602c498D216d680C4cAbb0Cd179d1d16",
         "multiaddrs":[
            "/dns4/bst-a01.test.keep.boar.network/tcp/4001"
         ],
         "network_id":"16Uiu2HAmMosdpAuRSw1ahNhqFq8e3Y4d4c5WZkjW1FGQi5WJwWZ7"
      },
      {
         "chain_address":"0xcAB2a402bAc470686d14956FB310D51BbEF9fA31",
         "multiaddrs":[
            "/dns4/bootstrap-1.test.keep.network/tcp/3919",
            "/dns4/keep-client-1.default.svc.cluster.local/tcp/3919"
         ],
         "network_id":"16Uiu2HAm3eJtyFKAttzJ85NLMromHuRg4yyum3CREMf6CHBBV6KY"
      },
      {
         "chain_address":"0x0eC14BC7cCA82c942Cf276F6BbD0413216dDB2bE",
         "multiaddrs":[
            "/dns4/bootstrap-0.test.keep.network/tcp/3919",
            "/dns4/keep-client-0.default.svc.cluster.local/tcp/3919"
         ],
         "network_id":"16Uiu2HAmCcfVpHwfBKNFbQuhvGuFXHVLQ65gB4sJm7HyrcZuLttH"
      },
      {
         "chain_address":"0xB8d1511aFab121f07cfDC7D1086371b1Fa758Fa2",
         "multiaddrs":[
            "/ip4/138.68.240.252/tcp/3919"
         ],
         "network_id":"16Uiu2HAm1buBFu8sHmgToezz6994uqQCZZ42azxrJP5GTxWmcBPA"
      }
   ]
}
```